src/components/ScrollViewWithContext.tsx

```diff
@@ -1,12 +1,21 @@
-import React, {useState, useRef} from 'react';
-import {ScrollView} from 'react-native';
+import React, {useState, useRef, ForwardedRef} from 'react';
+import {NativeScrollEvent, NativeSyntheticEvent, ScrollView} from 'react-native';
 
 const MIN_SMOOTH_SCROLL_EVENT_THROTTLE = 16;
 
-const ScrollContext = React.createContext();
+type ScrollContextValue = {
+    contentOffsetY?: number;
+    scrollViewRef?: ForwardedRef<ScrollView>;
+};
 
-// eslint-disable-next-line react/forbid-foreign-prop-types
-const propTypes = ScrollView.propTypes;
+const ScrollContext = React.createContext<ScrollContextValue>({});
```

Comment:
Always try to set a default value for context to avoid optional properties
```suggestion
type ScrollContextValue = {
    contentOffsetY: number;
    scrollViewRef: ForwardedRef<ScrollView>;
};

const ScrollContext = React.createContext<ScrollContextValue>({
    contentOffsetY: 0,
    scrollViewRef: null,
});
```